src/persist-client/src/batch.rs

```diff
@@ -131,17 +135,12 @@ where
     /// marks them as deleted.
     #[instrument(level = "debug", skip_all, fields(shard = %self.shard_id))]
     pub async fn delete(mut self) {
-        // TODO: This is temporarily disabled because nemesis seems to have
-        // caught that we sometimes delete batches that are later needed.
-        // Temporarily removing the deletions while we figure out the bug in
-        // case it has anything to do with CI timeouts.
-        //
-        // for key in self.blob_keys.iter() {
-        //     retry_external("batch::delete", || async {
-        //         self.blob.delete(key).await
-        //     })
-        //     .await;
-        // }
+        for part in self.batch.parts.iter() {
+            retry_external(&self.metrics.retries.external.batch_delete, || async {
+                self.blob.delete(&part.key).await
```

Comment:
thoughts on running this in a separate task so we don't block future writes on cleanup here?